### PR TITLE
[CIAPP] Document custom spans

### PR DESCRIPTION
**Pull request by edrevo**
### What does this PR do?
Add documentation for the custom span feature

### Preview

https://docs-staging.datadoghq.com/ximo/add-custom-spans-doc/continuous_integration/setup_pipelines/custom_spans/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

TODOs:
- [x] Blur identifiers in attached image
- [x] Release Jenkins plugin
- [x] Release datadog-ci
- [x] Release CircleCI support

### CODE REVIEW

File: content/en/continuous_integration/setup_pipelines/customspans.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+datadog-ci trace --name <name> <command>
+{{< /code-block >}}
+
+Specify a valid [Datadog API key][2] in the `DATADOG_API_KEY` environment variable. For example:
+
+{{< site-region region="us" >}}
+{{< code-block lang="bash" >}}
+DATADOG_API_KEY=<api_key> datadog-ci trace \
+  --service "Greet" \
+  echo "Hello World"
```
Comment (cg14823) [inline content/en/continuous_integration/setup_pipelines/customspans.md:43]: The block above this one says:

>  To trace a command, run the following:
> ```
> datadog-ci trace --name <name> <command>
> ```

Should the examples also include `--name` for consistency?

File: config/_default/menus/menus.en.yaml [hunk 2]
```diff
@@ -2006,6 +2006,11 @@ main:
     parent: setup_pipelines
     identifier: ci_jenkins
     weight: 202
+  - name: Custom spans
+    url: continuous_integration/setup_pipelines/customspans/
```
Comment (juan-fernandez) [inline config/_default/menus/menus.en.yaml:2010]: nit, but wouldn't this be more consistent? 
```suggestion
    url: continuous_integration/setup_pipelines/custom_spans/
```

**Review by cg14823 — COMMENTED** [review]
In the preview this page also shows for us-fed and us3 which we don't currently support I think for those pages we should show an appropriate message saying it isn't supported yet.

File: content/en/continuous_integration/setup_pipelines/customspans.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+Custom spans provide a way to trace individual commands in your CI pipelines, allowing you to measure the time your command takes without taking into account any setup or teardown actions that the job might have (for example, downloading Docker images or waiting for an available node in a Kubernetes-based infrastructure). These spans appear as part of the pipeline's trace:
+
+{{< img src="ci/ci-custom-spans.png" alt="Details for a single pipeline with custom spans" style="width:100%;">}}
+
+## Compatibility
+
+Custom spans work with the following CI providers:
+
+- Jenkins with Datadog plugin >= v3.2.0
+- CircleCI
```
Comment (edrevo) [inline content/en/continuous_integration/setup_pipelines/customspans.md:19]: We should remove this or wait until we merge CircleCI documentation before merging this.

File: content/en/continuous_integration/setup_pipelines/customspans.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+## Installing the Datadog CI CLI
+
+Install the [`datadog-ci`][1] CLI globally using `npm`:
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_pipelines/customspans.md:23]: I'd probably specify the version from which this is available (`>0.17.0`)

File: content/en/continuous_integration/setup_pipelines/customspans.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+{{< /site-region >}}
+
+## Configuration settings
+
+This is the full list of options available when using the `datadog-ci trace` command:
+
+`--name`
+: Display name of the custom span.<br/>
+**Default**: same value as `<command>`<br/>
+**Example**: `Wait for DB to be reachable`
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_pipelines/customspans.md:61]: do we want to show this between `"` ? 
Plus, is this the kind of value we want to encourage the user to use? This goes to operation name, doesn't it? These values are usually more like `grpc.client` or `junit.test`. Not that I don't think it wouldn't fit, just pointing it out.

**Review by juan-fernandez — APPROVED** [review]
LGTM, only minor stuff

File: content/en/continuous_integration/setup_pipelines/customspans.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
+---
+title: Custom spans
```
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/customspans.md:2]:
```suggestion
title: Adding custom spans to Pipeline Traces
```

File: content/en/continuous_integration/setup_pipelines/customspans.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+{{< code-block lang="bash" >}}
+datadog-ci trace --name <name> <command>
+{{< /code-block >}}
+
+Specify a valid [Datadog API key][2] in the `DATADOG_API_KEY` environment variable. For example:
+
+{{< site-region region="us" >}}
+{{< code-block lang="bash" >}}
+DATADOG_API_KEY=<api_key> datadog-ci trace \
+  --service "Greet" \
```
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/customspans.md:42]:
```suggestion
  --name "Greet" \
```

File: content/en/continuous_integration/setup_pipelines/customspans.md (hunk 5 above)
Comment (edrevo) [reply content/en/continuous_integration/setup_pipelines/customspans.md:61]: We are not using `"` in the junit page. I can add them in both places if you want.

This is not the operations name (which will be `<ciprovider>.custom`). This maps to the resource name.

**Review by cg14823 — APPROVED** [review]
👍

**Comment by fermayo** [comment]
@edrevo I think all dependencies are now met, correct? Is this unblocked?

**Comment by edrevo** [comment]
Yes, the PR is unblocked but it seems like the history of master was rewritten so this PR is a mess. Fixing it.

**Review by kayayarai — APPROVED** [review]

**Pull request merged**